File: samples/wordnik-api-java/src/main/java/com/wordnik/client/WordsApi.java
package com.wordnik.client;

import com.wordnik.client.ApiException;
import com.wordnik.client.ApiInvoker;
import java.util.Date;
import com.wordnik.client.model.Definition;
import com.wordnik.client.model.ContentProvider;
import com.wordnik.client.model.Example;
import com.wordnik.client.model.WordObject;
import com.wordnik.client.model.WordOfTheDay;
import com.wordnik.client.model.WordFrequency;
import java.util.*;

public class WordsApi {
  String basePath = "http://api.wordnik.com/v4";
  ApiInvoker apiInvoker = ApiInvoker.getInstance();

  public ApiInvoker getInvoker() {
    return apiInvoker;
  }
  
  public void setBasePath(String basePath) {
    this.basePath = basePath;
  }
  
  public String getBasePath() {
    return basePath;
  }

  public List<WordFrequency> searchWords (String query, String includePartOfSpeech, String excludePartOfSpeech, int minCorpusCount, int maxCorpusCount, int minDictionaryCount, int maxDictionaryCount, int minLength, int maxLength, int skip, int limit, String caseSensitive) throws ApiException {
    // create path and map variables
    String path = "/words.{format}/search".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    // verify required params are set
    if(query == null ) {
       throw new ApiException(400, "missing required params");
    }
    if(!"null".equals(String.valueOf(query)))
      queryParams.put("query", String.valueOf(query));
    if(!"null".equals(String.valueOf(caseSensitive)))
      queryParams.put("caseSensitive", String.valueOf(caseSensitive));
    if(!"null".equals(String.valueOf(includePartOfSpeech)))
      queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
    if(!"null".equals(String.valueOf(excludePartOfSpeech)))
      queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
    if(!"null".equals(String.valueOf(minCorpusCount)))
      queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
    if(!"null".equals(String.valueOf(maxCorpusCount)))
      queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
    if(!"null".equals(String.valueOf(minDictionaryCount)))
      queryParams.put("minDictionaryCount", String.valueOf(minDictionaryCount));
    if(!"null".equals(String.valueOf(maxDictionaryCount)))
      queryParams.put("maxDictionaryCount", String.valueOf(maxDictionaryCount));
    if(!"null".equals(String.valueOf(minLength)))
      queryParams.put("minLength", String.valueOf(minLength));
    if(!"null".equals(String.valueOf(maxLength)))
      queryParams.put("maxLength", String.valueOf(maxLength));
    if(!"null".equals(String.valueOf(skip)))
      queryParams.put("skip", String.valueOf(skip));
    if(!"null".equals(String.valueOf(limit)))
      queryParams.put("limit", String.valueOf(limit));
    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams);
      if(response != null){
        return (List<WordFrequency>) ApiInvoker.deserialize(response, "List", WordFrequency.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
      	return null;
      }
      else {
        throw ex;
      }
    }
  }
  public WordOfTheDay getWordOfTheDay (String date) throws ApiException {
    // create path and map variables
    String path = "/words.{format}/wordOfTheDay".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    if(!"null".equals(String.valueOf(date)))
      queryParams.put("date", String.valueOf(date));
    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams);
      if(response != null){
        return (WordOfTheDay) ApiInvoker.deserialize(response, "", WordOfTheDay.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
      	return null;
      }
      else {
        throw ex;
      }
    }
  }
  public List<WordObject> getRandomWords (String hasDictionaryDef, String includePartOfSpeech, String excludePartOfSpeech, int minCorpusCount, int maxCorpusCount, int minDictionaryCount, int maxDictionaryCount, int minLength, int maxLength, String sortBy, String sortOrder, int limit) throws ApiException {
    // create path and map variables
    String path = "/words.{format}/randomWords".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    if(!"null".equals(String.valueOf(hasDictionaryDef)))
      queryParams.put("hasDictionaryDef", String.valueOf(hasDictionaryDef));
    if(!"null".equals(String.valueOf(includePartOfSpeech)))
      queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
    if(!"null".equals(String.valueOf(excludePartOfSpeech)))
      queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
    if(!"null".equals(String.valueOf(minCorpusCount)))
      queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
    if(!"null".equals(String.valueOf(maxCorpusCount)))
      queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
    if(!"null".equals(String.valueOf(minDictionaryCount)))
      queryParams.put("minDictionaryCount", String.valueOf(minDictionaryCount));
    if(!"null".equals(String.valueOf(maxDictionaryCount)))
      queryParams.put("maxDictionaryCount", String.valueOf(maxDictionaryCount));
    if(!"null".equals(String.valueOf(minLength)))
      queryParams.put("minLength", String.valueOf(minLength));
    if(!"null".equals(String.valueOf(maxLength)))
      queryParams.put("maxLength", String.valueOf(maxLength));
    if(!"null".equals(String.valueOf(sortBy)))
      queryParams.put("sortBy", String.valueOf(sortBy));
    if(!"null".equals(String.valueOf(sortOrder)))
      queryParams.put("sortOrder", String.valueOf(sortOrder));
    if(!"null".equals(String.valueOf(limit)))
      queryParams.put("limit", String.valueOf(limit));
    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams);
      if(response != null){
        return (List<WordObject>) ApiInvoker.deserialize(response, "List", WordObject.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
      	return null;
      }
      else {
        throw ex;
      }
    }
  }
  public WordObject getRandomWord (String includePartOfSpeech, String excludePartOfSpeech, int minCorpusCount, int maxCorpusCount, int minDictionaryCount, int maxDictionaryCount, int minLength, int maxLength, String hasDictionaryDef) throws ApiException {
    // create path and map variables
    String path = "/words.{format}/randomWord".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();

    if(!"null".equals(String.valueOf(hasDictionaryDef)))
      queryParams.put("hasDictionaryDef", String.valueOf(hasDictionaryDef));
    if(!"null".equals(String.valueOf(includePartOfSpeech)))
      queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
    if(!"null".equals(String.valueOf(excludePartOfSpeech)))
      queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
    if(!"null".equals(String.valueOf(minCorpusCount)))
      queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
    if(!"null".equals(String.valueOf(maxCorpusCount)))
      queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
    if(!"null".equals(String.valueOf(minDictionaryCount)))
      queryParams.put("minDictionaryCount", String.valueOf(minDictionaryCount));
    if(!"null".equals(String.valueOf(maxDictionaryCount)))
      queryParams.put("maxDictionaryCount", String.valueOf(maxDictionaryCount));
    if(!"null".equals(String.valueOf(minLength)))
      queryParams.put("minLength", String.valueOf(minLength));
    if(!"null".equals(String.valueOf(maxLength)))
      queryParams.put("maxLength", String.valueOf(maxLength));
    try {
      String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, null, headerParams);
      if(response != null){
        return (WordObject) ApiInvoker.deserialize(response, "", WordObject.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      if(ex.getCode() == 404) {
      	return null;
      }
      else {
        throw ex;
      }
    }
  }
  }

File: samples/wordnik-api-scala/src/main/scala/com/wordnik/client/WordsApi.scala
package com.wordnik.client

import com.wordnik.client.ApiInvoker
import com.wordnik.client.ApiException

import java.util.Date
import com.wordnik.client.model.Definition
import com.wordnik.client.model.ContentProvider
import com.wordnik.client.model.Example
import com.wordnik.client.model.WordObject
import com.wordnik.client.model.WordSearchResult
import com.wordnik.client.model.WordOfTheDay
import com.wordnik.client.model.WordFrequency
import com.wordnik.client.model.WordSearchResults
import scala.collection.mutable.HashMap

class WordsApi {
  var basePath: String = "http://api.wordnik.com/v4"
  var apiInvoker = ApiInvoker

  def searchWords (query: String, includePartOfSpeech: String, excludePartOfSpeech: String, maxCorpusCount: String, maxDictionaryCount: String, maxLength: String, caseSensitive: String= "true", minCorpusCount: String= "5", minDictionaryCount: String= "1", minLength: String= "1", skip: String= "0", limit: String= "10") : Option[List[WordFrequency]]= {
    // create path and map variables
    val path = "/words.{format}/search".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    // verify required params are set
    (Set(query) - null).size match {
       case 1 => // all required values set
       case _ => throw new Exception("missing required params")
    }
    if(String.valueOf(query) != "null") queryParams += "query" -> query.toString
    if(String.valueOf(caseSensitive) != "null") queryParams += "caseSensitive" -> caseSensitive.toString
    if(String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if(String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if(String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if(String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if(String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if(String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if(String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if(String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    if(String.valueOf(skip) != "null") queryParams += "skip" -> skip.toString
    if(String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "List", classOf[WordFrequency]).asInstanceOf[List[WordFrequency]])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def getWordOfTheDay (date: String, category: String, creator: String) : Option[WordOfTheDay]= {
    // create path and map variables
    val path = "/words.{format}/wordOfTheDay".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    if(String.valueOf(date) != "null") queryParams += "date" -> date.toString
    if(String.valueOf(category) != "null") queryParams += "category" -> category.toString
    if(String.valueOf(creator) != "null") queryParams += "creator" -> creator.toString
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordOfTheDay]).asInstanceOf[WordOfTheDay])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def searchWordsNew (query: String, includePartOfSpeech: String, excludePartOfSpeech: String, maxCorpusCount: String, maxDictionaryCount: String, maxLength: String, caseSensitive: String= "true", minCorpusCount: String= "5", minDictionaryCount: String= "1", minLength: String= "1", skip: String= "0", limit: String= "10") : Option[WordSearchResults]= {
    // create path and map variables
    val path = "/words.{format}/search/{query}".replaceAll("\\{format\\}","json").replaceAll("\\{" + "query" + "\\}",apiInvoker.escapeString(query))

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    // verify required params are set
    (Set(query) - null).size match {
       case 1 => // all required values set
       case _ => throw new Exception("missing required params")
    }
    if(String.valueOf(caseSensitive) != "null") queryParams += "caseSensitive" -> caseSensitive.toString
    if(String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if(String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if(String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if(String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if(String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if(String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if(String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if(String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    if(String.valueOf(skip) != "null") queryParams += "skip" -> skip.toString
    if(String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordSearchResults]).asInstanceOf[WordSearchResults])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def getRandomWords (hasDictionaryDef: String, includePartOfSpeech: String, excludePartOfSpeech: String, minCorpusCount: String, maxCorpusCount: String, minDictionaryCount: String, maxDictionaryCount: String, minLength: String, maxLength: String, sortBy: String, sortOrder: String, limit: String) : Option[List[WordObject]]= {
    // create path and map variables
    val path = "/words.{format}/randomWords".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    if(String.valueOf(hasDictionaryDef) != "null") queryParams += "hasDictionaryDef" -> hasDictionaryDef.toString
    if(String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if(String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if(String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if(String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if(String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if(String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if(String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if(String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    if(String.valueOf(sortBy) != "null") queryParams += "sortBy" -> sortBy.toString
    if(String.valueOf(sortOrder) != "null") queryParams += "sortOrder" -> sortOrder.toString
    if(String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "List", classOf[WordObject]).asInstanceOf[List[WordObject]])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  def getRandomWord (includePartOfSpeech: String, excludePartOfSpeech: String, minCorpusCount: String, maxCorpusCount: String, minDictionaryCount: String, maxDictionaryCount: String, minLength: String, maxLength: String, hasDictionaryDef: String= "true") : Option[WordObject]= {
    // create path and map variables
    val path = "/words.{format}/randomWord".replaceAll("\\{format\\}","json")// query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]

    if(String.valueOf(hasDictionaryDef) != "null") queryParams += "hasDictionaryDef" -> hasDictionaryDef.toString
    if(String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if(String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if(String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if(String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if(String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if(String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if(String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if(String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordObject]).asInstanceOf[WordObject])
        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }
  }
